### PR TITLE
add trim, ltrim, rtrim functions

### PR DESCRIPTION
**Pull request by agonist**
Add the trim function and its 2 variants ltrim rtrim.

### CODE REVIEW

File: internal/expr/functions/definition.go [hunk 1]
```diff
@@ -57,13 +57,20 @@ func (fd *definition) Name() string {
...
+
 	if len(args) != fd.arity {
 		return nil, fmt.Errorf("%s() takes %d argument(s), not %d", fd.name, fd.arity, len(args))
 	}
 	return fd.constructorFn(args...)
 }
 
 func (fd *definition) String() string {
+	if fd.arity < 0 {
+		fd.arity = 0
```
Comment (asdine) [inline internal/expr/functions/definition.go:72]: I wouldn't change the arity here. Instead we could use a local variable
```go
arity := fd.arity
if arity < 0 {
...
```

File: internal/expr/functions/strings.go [hunk 2]
```diff
@@ -109,3 +130,72 @@ func (s *Upper) Params() []expr.Expr { return []expr.Expr{s.Expr} }
...
+		}
+		cutset = types.As[string](remove)
+	}
+
+	trimmed := s.TrimFunc(types.As[string](input), cutset)
+
+	return types.NewTextValue(trimmed), nil
+}
+
+func (s *Trim) IsEqual(other []expr.Expr) bool {
```
[the remark this answers is not in the document]
Comment (agonist) [reply internal/expr/functions/strings.go]: fixed

File: internal/expr/functions/definition.go (hunk 1 above)
Comment (agonist) [reply internal/expr/functions/definition.go]: fixed

**Review by asdine — APPROVED** [review]
👏🏼

**Pull request merged**